### ISSUE TITLE
Suggestion / discussion

### ISSUE THREAD

**Issue by vikyw89** [zhudotexe/redel#5]
I love the idea, haven't look at the code but. What do you think of refining the break case for the recursion as undivideable task ? This to prevent delegation of a single task.

A task that can no longer be divided into subtasks cannot be delegated.

Pretty sure you have thought about it, what problem did you encountered when using that approach ?

**Comment by zhudotexe** [comment]
Yeah, the happy path is when the LLM decides not to decompose a simple-enough task. The biggest challenge is deciding when a task is "simple enough" - most LLMs don't have a good sense of what they do/don't "know" so they tend to overcommit to a too complex task or undercommit (see §5 of [our paper](https://aclanthology.org/2024.emnlp-demo.17.pdf)). It might be solvable using better prompting (though this requires more human labor), or on demand if a node fails a subtask and you have an oracle (e.g., https://arxiv.org/abs/2311.05772).
